core/templates/dev/head/components/question-directives/questions-list/questions-list.directive.ts

```diff
@@ -428,6 +457,100 @@ angular.module('oppia').directive('questionsList', [
             }
           };
 
+          ctrl.changeDifficulty = function(
+              questionId, skillDescriptions, skillDifficulties) {
+            if (!ctrl.canEditQuestion()) {
+              AlertsService.addWarning(
+                'User does not have enough rights to edit the question');
+              return;
+            }
+            var linkedSkillsWithDifficulty = [];
+            if (ctrl.getAllSkillSummaries().length === 0) {
+              linkedSkillsWithDifficulty.push(
+                SkillDifficultyObjectFactory.create(
+                  ctrl.skillIds[0], skillDescriptions[0], skillDifficulties[0])
```

Comment:
skillSummaries is only used by the topic editor so that the ID of the skills can be obtained using their descriptions, whereas `skillIds` is used to fetch all questions for all skills in a topic.
In the skill editor, there is only one ID, hence skillSummaries is empty (it's not needed), but skillIds still has the one element.